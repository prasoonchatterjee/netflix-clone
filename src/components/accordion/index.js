import React, { useState, useContext, createContext } from "react";
import {
  Container,
  Title,
  Frame,
  Item,
  Inner,
  Header,
  Body,
} from "./styles/accordion";
const ToggleContext = createContext();

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggleShow, setToggeleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggeleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  const { toggleShow, setToggeleShow } = useContext(ToggleContext);
  return (
    <Header
      onClick={() => setToggeleShow((toggleShow) => !toggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...restProps }) {
  const { toggleShow } = useContext(ToggleContext);

  return toggleShow && <Body {...restProps}>{children}</Body>;
};
