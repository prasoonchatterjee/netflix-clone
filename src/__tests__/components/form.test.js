import React from "react";
import { render } from "@testing-library/react";
import { Form } from "../../components";

jest.mock("react-router-dom");

describe("<Form />", () => {
  test("renders <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>
        <Form.Base>
          <Form.Input placeholder="Email Address" onChange={() => {}} />
          <Form.Input
            placeholder="Password"
            onChange={() => {}}
            type="password"
          />
          <Form.Submit type="submit" disabled>
            Sign In
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          New to Netflix?
          {/* <Form.Link to="/signup">Sign up now.</Form.Link> */}
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a bot.
          Learn more.
        </Form.TextSmall>
      </Form>
    );

    expect(
      getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
      )
    ).toBeTruthy();
    expect(getByText("Sign In Now")).toBeTruthy();
    expect(getByText("Sign In")).toBeTruthy();
    expect(getByText("Sign In").disabled).toBeTruthy();
    expect(getByText("New to Netflix?")).toBeTruthy();
    expect(getByPlaceholderText("Email Address")).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders <Form /> with an error", () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Your email address is already being used</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );

    expect(getByText("Your email address is already being used")).toBeTruthy();
    expect(queryByText("Sign In").disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
