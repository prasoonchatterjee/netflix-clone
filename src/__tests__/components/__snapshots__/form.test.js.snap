// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders <Form /> with an error 1`] = `
<div
  class="form__Container-sc-1jfxee4-0 drnOcQ"
>
  <div
    class="form__Error-sc-1jfxee4-1 eITLaO"
  >
    Your email address is already being used
  </div>
  <button
    class="form__Submit-sc-1jfxee4-8 kJmNGK"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form /> renders <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-1jfxee4-0 drnOcQ"
>
  <h1
    class="form__Title-sc-1jfxee4-3 ctOVSL"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-1jfxee4-2 bFvRxs"
  >
    <input
      class="form__Input-sc-1jfxee4-7 ZBIYB"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-1jfxee4-7 ZBIYB"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-1jfxee4-8 kJmNGK"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-1jfxee4-4 gSzvgS"
  >
    New to Netflix?
  </p>
  <p
    class="form__TextSmall-sc-1jfxee4-5 bhLOOK"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
